lib/rspec/mocks/proxy.rb

```diff
@@ -434,6 +440,11 @@ def warn(method_name)
         source = CallerFilter.first_non_rspec_line
         Kernel.warn("An expectation of :#{method_name} was set on nil. Called from #{source}. Use allow_message_expectations_on_nil to disable warnings.")
       end
+
+      def raise_error(method_name)
+        source = CallerFilter.first_non_rspec_line
+        raise("An expectation of :#{method_name} was set on nil. Called from #{source}. To allow expectations on nil, set disallow_expectations_on_nil to false.")
```

Comment:
@myronmarston or @JonRowe  will chime in :)

Grepping around I found https://github.com/rspec/rspec-mocks/blob/master/lib/rspec/mocks/error_generator.rb which suggests two things:

1. We do use custom exception classes, but they inherit from `StandardError` rather than a common mocks type. However, we also typically use `ArgumentError` and `NoMethodError` (ruby built-ins), either of which is almost applicable here. Also found a single example of raising `StandardError` at https://github.com/rspec/rspec-mocks/blob/master/lib/rspec/mocks/mutate_const.rb#L329 so...
2. We may want to use `ErrorGenerator`, which delegates to `RSpec::Support.notify_failure`. We use that method to support things like aggregate failures (I think). [We (rspec team) outside of this PR should probably add some class doc to explain why that class exists and when it is/isn't appropriate to use.] This isn't used everywhere, but I think it's a fairly recent addition and just not everything is ported to it yet maybe? It appears to handle similar types of failures to the one you're adding here.